packages/jellyfish-api-core/__tests__/category/loan/placeAuctionBid.test.ts

```diff
@@ -463,6 +463,86 @@ describe('placeAuctionBid success', () => {
+
+  it('should make the vault active after completion of one auction batch', async () => {
+    // update price oracle
+    await alice.rpc.oracle.setOracleData(oracleId, now(), { prices: [{ tokenAmount: '100@DFI', currency: 'USD' }, { tokenAmount: '100@BTC', currency: 'USD' }, { tokenAmount: '100@TSLA', currency: 'USD' }, { tokenAmount: '1@AMZN', currency: 'USD' }] })
+    await alice.generate(12)
+
+    // Loan200 scheme set up
+    await alice.rpc.loan.createLoanScheme({
+      minColRatio: 200,
+      interestRate: new BigNumber(1),
+      id: 'Loan200'
+    })
+    await alice.generate(1)
+
+    // create another vault
+    const aliceVaultAddr = await alice.generateAddress()
+    const aliceVaultId = await alice.rpc.loan.createVault({
+      ownerAddress: aliceVaultAddr,
+      loanSchemeId: 'Loan200'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.loan.depositToVault({
+      vaultId: aliceVaultId, from: aliceColAddr, amount: '60@DFI'
+    })
+    await alice.generate(1)
+
+    await alice.rpc.loan.depositToVault({
+      vaultId: aliceVaultId, from: aliceColAddr, amount: '60@BTC'
+    })
+    await alice.generate(1)
+
+    const aliceLoanAddr = await bob.generateAddress()
+    await alice.rpc.loan.takeLoan({
+      vaultId: aliceVaultId,
+      amounts: '60@TSLA',
+      to: aliceLoanAddr
+    })
+    await alice.generate(1)
+
+    const aliceLoanAcc = await bob.container.call('getaccount', [aliceLoanAddr])
+    expect(aliceLoanAcc).toStrictEqual(['60.00000000@TSLA'])
+
+    // update the price again for 101@TSLA
+    await alice.rpc.oracle.setOracleData(oracleId, now(), { prices: [{ tokenAmount: '100@DFI', currency: 'USD' }, { tokenAmount: '100@BTC', currency: 'USD' }, { tokenAmount: '101@TSLA', currency: 'USD' }, { tokenAmount: '1@AMZN', currency: 'USD' }] })
+    await alice.generate(12)
+
+    // see the aliceVaultId inLiquidation
+    const vault = await alice.rpc.loan.getVault(aliceVaultId) as VaultLiquidation
+    expect(vault.state).toStrictEqual('inLiquidation')
+    expect(vault.batches).toStrictEqual([
+      { index: 0, collaterals: ['49.99999980@DFI', '49.99999980@BTC'], loan: '50.00011389@TSLA' },
+      { index: 1, collaterals: ['10.00000020@DFI', '10.00000020@BTC'], loan: '10.00002303@TSLA' }
+    ])
+
+    // place auction bid for the first batch
+    const txid1 = await alice.rpc.loan.placeAuctionBid({
+      vaultId: aliceVaultId,
+      index: 0,
+      from: aliceColAddr,
+      amount: '59.41@TSLA'
+    })
+    expect(typeof txid1).toStrictEqual('string')
+    await alice.container.generate(1)
+
+    // reduce the TSLA price to 99. by the time auction ends, vault will have enough col ratio to survive reliquidation.
+    await alice.rpc.oracle.setOracleData(oracleId, now(), { prices: [{ tokenAmount: '100@DFI', currency: 'USD' }, { tokenAmount: '100@BTC', currency: 'USD' }, { tokenAmount: '99@TSLA', currency: 'USD' }, { tokenAmount: '1@AMZN', currency: 'USD' }] })
```

Comment:
i see.. actually the core thing is the price changes on collateral ratio.. the bid is doesn't matter
loan price x100 -> vault liquidated -> loan drop /1000 -> vault recovered after auction end without bid

Reply:
yes, it could happen like that as well. both are valid use cases.
I will take the remaining of https://github.com/DeFiCh/jellyfish/pull/722#issuecomment-975817416 separately. 
We can merge this PR I think.